streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -215,12 +215,12 @@ public long size() {
     }
 
     /**
-     * Reject late events that arrive more than {@code afterWindowEnd}
+     * Reject out-of-order events that arrive more than {@code afterWindowEnd}
      * after the end of its window.
      *
-     * Lateness is defined as (stream_time - record_timestamp).
+     * Delay is defined as (stream_time - record_timestamp).
```

Comment:
This is now defining a term that is unused elsewhere. We can either define the term we use ("out-of-ordered-ness" is defined as...") or we can use the term we define (above: "Reject out-of-order events that are delayed more than...").

Reply:
I added a sentence to `Windows` JavaDoc -- do you think that is sufficient? Don't want to make the JavaDocs too complicated.